#!/usr/bin/env python

"""
 MD25Controller.cfg

 Copyright (C) 2018 Alessandro Francescon
 
 This program is free software: you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation, either version 3 of the License.
 
 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.
 
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
"""

PACKAGE = "md25"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("controller_linear_gain", double_t, 0, "The controlle PID gain for linear velocity", 10.0, 0, 100)
gen.add("controller_linear_derivative_gain", double_t, 0, "The controlle PID derivative gain for linear velocity", 10.0, 0, 100)
gen.add("controller_angular_gain", double_t, 0, "The controlle PID gain for angular velocity", 10.0, 0, 100)
gen.add("controller_angular_derivative_gain", double_t, 0, "The controlle PID derivative gain for angular velocity", 10.0, 0, 100)

exit(gen.generate(PACKAGE, PACKAGE, "MD25Controller"))
